### ISSUE TITLE
Create a simple utility for VA Energy staff to upload data files, esp. for energy efficiency data

### ISSUE THREAD

**Issue by arthursmalliii** [coopercenter/va_clean_economy_dashboard#31]
As discussed. Make it dead simple, with a stable URL. 

Need some simple login/security, so bots don't upload malware for us.

**Comment by arthursmalliii** [comment]
See: https://googledrive.tidyverse.org/

**Comment by arthursmalliii** [comment]
It will be better if they load the files on their own cloud file storage system, then we fetch it from there, rather than have them upload to a UVA file storage system. Getting us "read" permission to their system will likely be easier than getting them "write" permissions to ours.

**Comment by caneale320** [comment]
@arthursmalliii A few preliminary notes:

For security purposes I'd suggest we create (or use, if extant) a dummy Gmail account which Bettina could authorize to access a specific folder, such that only that account could access the folder and we don't have to make the link publicly accessible. Even if we did have the link publicly accessible, it would be read only and I don't think would pose much of a threat.

This package, and google drive downloads in general, use a unique file ID to execute downloads. Because of this we can:

1. update data without altering this unique ID by pasting over the old data with new data, not uploading a new file. I haven't been able to find another way of using a stable URL for just the file.
2. use a folder as a stable URL and use the most recent file in that folder. 
3. use an agreed upon naming convention to identify the file in a given folder. We could request that the provided data be the only file in the folder, then implement additional checks (filetype, upload date, sharing permissions, actual data content, etc) to make sure it is as we expect before integrating it into the database/dashboard. No stable URL, but checks in place to assure data integrity.

Option one is definitely a smaller lift on our end, but seems brittle (they accidently delete the file, paste incorrectly, etc). Option two is similarly a small lift and eliminates pasting errors but is susceptible to other errors. Option three seems the most robust to me, but would be the least "simple", though I don't think crazy. 

In any case, I can contact Bettina and ask her about getting permissions to a specific folder on their drive and what that would entail. 

Let me know your thoughts and how you'd like me to proceed.

**Comment by caneale320** [comment]
@bettinabergoo I've sent you an invite to be a collaborator on a test Box Folder. (Folder link: https://virginia.box.com/s/3ntsukrcr4tt173n6pnw22j9rzid4s3n). Could you please make sure the permissions are set up correctly such that you can upload a file by placing a test spreadsheet in the folder?

Thanks.

**Comment by bettinabergoo** [comment]
@caneale320 Done -- can you access?

**Comment by caneale320** [comment]
@bettinabergoo Yup! Beautiful. Thanks! I'll get you a link and invite to the final location soon, but at least we know that Box will let us do what we need to. 

@arthursmalliii Could you please provide a Weldon Cooper Center folder or in your UVA Box, whatever you prefer, as the permanent location for the folder?

**Comment by arthursmalliii** [comment]
@caneale320 : [Folder created.](https://virginia.box.com/s/lvgdvdbccguv89g3wx7fhpezul4m6i1h) You have editor rights. Please take it from here.

Does Box have an API that will work for this use case?

**Comment by caneale320** [comment]
@arthursmalliii Yup! Will test it today/tomorrow and report results. It has [native support for Python](https://developer.box.com/guides/tooling/sdks/python/) which would be my preference but there appears to be third party support through [an R package](https://r-box.github.io/boxr/index.html) if you'd like to stick to one language for your own/others' use I imagine I could make it work, though I haven't looked into the R package much yet.

**Comment by arthursmalliii** [comment]
Python is fine.

Sent from my mobile. Apologies for brevity and typos.
________________________________

**Comment by caneale320** [comment]
UVA's box settings are set up such that I have to get approval to make API requests. I've sent an approval request through the box portal and a follow up email to the listed box admin email, copied @arthursmalliii
